api/delete.go

```diff
@@ -34,5 +38,13 @@ func (d *serviceDeletor) Delete(serviceID string) error {
 	if err := s.Stop(); err != nil {
 		return err
 	}
+	// delete volumes first before the service. this way if
+	// deleting volumes fails, process can be retried by the user again
+	// because service still will be in the db.
+	if deleteData {
```

Comment:
Why not always deleting the volumes?
If an user delete a service, it's natural to delete the associated volume.

Reply:
In the real world usage data should not be deleted automatically without the confirmation. Let's think that we have a MESG service that manages passwords. And this service is used by many people. But after a while we decided to shutdown this service by deleting it. Still, passwords database should be kept to give a copy of them to their owners let's say via email. In a case like this, deleting a service not always means that data should be deleted as well.

Of course instead of removing the service, devs can stop it and keep the data but data is very sensitive thing so we should avoid any margin to errors. Since this is a UX thing and there is no one right answer, I'll not push on to have this delete optionally but this is what I'm thinking.

Second reason why we need this option is: In active development while using aliases, we'll want to keep data volumes. In order to do that within _dev_ command we should have this option to not delete data volumes. And for the devs who doesn't use _dev_ and just use the start command, when they update their source code they'll naturally use the delete command  directly to stop and delete the service -instead of stop-. So it shouldn't automatically delete the volume because it'll be needed during development.

Reply:
Good point

Reply:
What about services that have the same alias? Like a update of a service.
If the user want to delete the previous version (with the service ID), in your case, it will delete the volume that are actually used by the latest version of the service.
We need to leave this option to the user.

Reply:
Otherwise we could consider that if it’s an I’d then we don’t delete, if it’s an alias we delete. The volume is related to an alias not an id so it makes sense. I’m just afraid that if even for us it looks complicated with the different flags it means it will be way too complicated for users.

Reply:
I've pushed a new commit (implemented https://github.com/mesg-foundation/core/pull/613#discussion_r238206929). Please check and let's have a final decision.

Reply:
* _-y_ answers every question as yes. With `mesg-core service -y --all`, all services and their data will be deleted with no question asked.